pymeasure/instruments/instrument.py

```diff
@@ -244,36 +246,103 @@ def id(self):
             raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
 
     # Wrapper functions for the Adapter object
-    def ask(self, command):
-        """ Writes the command to the instrument through the adapter
-        and returns the read response.
+    def write(self, command, **kwargs):
+        """Write a string command to the instrument appending `write_termination`.
 
         :param command: command string to be sent to the instrument
+        :param kwargs: Keyword arguments for the adapter.
         """
-        return self.adapter.ask(command)
+        self.adapter.write(command, **kwargs)
 
-    def write(self, command):
-        """ Writes the command to the instrument through the adapter.
+    def write_bytes(self, content, **kwargs):
+        """Write the bytes `content` to the instrument."""
+        self.adapter.write_bytes(content, **kwargs)
 
-        :param command: command string to be sent to the instrument
+    def read(self, **kwargs):
+        """Read up to (excluding) `read_termination` or the whole read buffer."""
+        return self.adapter.read(**kwargs)
+
+    def read_bytes(self, count, **kwargs):
+        """Read a certain number of bytes from the instrument.
+
+        :param int count: Number of bytes to read. A value of -1 indicates to
+            read the whole read buffer.
+        :param kwargs: Keyword arguments for the adapter.
+        :returns bytes: Bytes response of the instrument (including termination).
         """
-        self.adapter.write(command)
+        return self.adapter.read_bytes(count, **kwargs)
 
-    def read(self):
-        """ Reads from the instrument through the adapter and returns the
-        response.
+    def write_binary_values(self, command, values, *args, **kwargs):
+        """Write binary values to the device.
+
+        :param command: Command to send.
+        :param values: The values to transmit.
+        :param \\*args, \\**kwargs: Further arguments to hand to the Adapter.
         """
-        return self.adapter.read()
+        self.adapter.write_binary_values(command, values, *args, **kwargs)
+
+    # Communication functions
+    def wait_until_read(self, query_delay=0):
```

Comment:
Sorry for the last minute iteration, but would not that be better/clearer called `wait_for(delay)`? The way it ended up being implemented, it only does a sleep, no reading, no query involved.

Reply:
Also, in the docstring maybe add a remark that this is used automatically in the `ask` method.

Reply:
done